protos/protos/user.proto

```diff
@@ -8,28 +8,20 @@ message Problem {
 	uint32 id = 1;
 	string name = 2;
 	string description = 3; // long name
-	repeated uint32 testcases_per_subtask = 4;
-	common.Resources runtime_limits = 5;
-	uint64 source_size_limit = 6;
-	string type = 7; // unsure about this
+	// repeated uint32 testcases_per_subtask = 4;
+	// common.Resources runtime_limits = 5;
+	// uint64 source_size_limit = 6;
+	// string type = 7; // unsure about this
 	// available languages ...
 	// statements & various files ...
 }
 
-message Contest {
-	string name = 1;
-	string description = 2;
-	google.protobuf.Timestamp start_time = 3;
-	google.protobuf.Timestamp end_time = 4;
-	optional uint32 submission_limit = 5;
-	//uint32 problem_count;
-	repeated Problem problem_infos = 6; // this better?
-}
-
 message Message { // questions and announcements are the same
-	string subject = 1;
-	optional uint32 problem_id = 2;
-	string text = 3;
-	optional string user = 4;
-	google.protobuf.Timestamp timestamp = 5;
+	uint32 id = 1;
+	string subject = 2;
+	optional uint32 problem_id = 3;
+	string text = 4;
+	optional string from = 5;
+	optional string to = 6;
+	google.protobuf.Timestamp timestamp = 7;
```

Comment:
Personally, I would call this `creation_timestamp` or `created_at` or similar instead of the general `timestamp` which does not explicitly say what this timestamp means.